src/com/amazon/ionschema/IonSchemaSystemBuilder.kt

```diff
@@ -93,7 +95,28 @@ class IonSchemaSystemBuilder private constructor() {
      */
     @Deprecated("For backwards compatibility with v1.0")
     fun allowAnonymousTopLevelTypes(): IonSchemaSystemBuilder {
-        params.put(IonSchemaSystemImpl.Param.ALLOW_ANONYMOUS_TOP_LEVEL_TYPES, Object())
+        params.put(IonSchemaSystemImpl.Param.ALLOW_ANONYMOUS_TOP_LEVEL_TYPES, true)
+        return this
+    }
+
+    /**
+     * Allows forward-compatibility with the fixed, spec-compliant behavior for handling
+     * schema imports that is introduced in `ion-schema-kotlin-2.0.0`.
+     *
+     * @since 1.2
+     */
+    fun allowTransitiveImports(boolean: Boolean): IonSchemaSystemBuilder {
+        params[IonSchemaSystemImpl.Param.ALLOW_TRANSITIVE_IMPORTS] = boolean
+        return this
+    }
+
+    /**
+     * Provides a callback for the IonSchemaSystem to use for writing log messages.
+     *
+     * @since 1.2
+     */
+    fun withLogger(logger: IonSchemaSystemLogger): IonSchemaSystemBuilder {
```

Comment:
The use of the term "log" and "logger" throughout this PR feels too specific to me. I'd rather just call it a callback and state in the documentation that one of the use cases for the callback is to allow the user to log the message. But all this is really doing is allowing the user to receive a string message from the library that something bad (but currently non-exceptional) happened. The user doesn't have to log this; they can ignore it, throw their own error, etc.